ext/symengine/ruby_basic.c

```diff
@@ -32,28 +32,37 @@ VALUE cbasic_binary_op(VALUE self, VALUE operand2,
     sympify(operand2, cbasic_operand2);
 
     cresult = basic_new_heap();
-    cwfunc_ptr(cresult, this, cbasic_operand2);
-    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL, cbasic_free_heap,
-                              cresult);
-    basic_free_stack(cbasic_operand2);
 
-    return result;
+    int error_code = cwfunc_ptr(cresult, this, cbasic_operand2);
+    if (error_code == 0) {
+        result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL,
+                                  cbasic_free_heap, cresult);
+        basic_free_stack(cbasic_operand2);
+        return result;
+    } else {
+        basic_free_stack(cbasic_operand2);
+        rb_raise(rb_eRuntimeError, "Runtime Error");
```

Comment:
@v0dro this is in place until a exceptions are made in the SymEngine C++ code (right now it's all `std::runtime_error`s). Anyway I have written the code to be integrated into the cwrappers to catch multiple exceptions (for a method like `parse` this is needed). Then depending on the error code sent, the suitable error message will be displayed. This is still WIP. Please check https://github.com/symengine/symengine/pull/1044 for the discussion on this :)